lib/extendedkey.ex

```diff
@@ -72,25 +89,36 @@ defmodule Bitcoinex.ExtendedKey do
     end
 
     defp str_to_level(level) do
-      [num | tick] = String.split(level, ["'", "h"])
+      {num, is_hardened} =
+        case String.split(level, ["'", "h"]) do
+          [num] ->
+            {num, false}
+
+          [num, ""] ->
+            {num, true}
+        end
+
       nnum = String.to_integer(num)
 
-      if nnum >= @softcap or nnum < 0 do
-        raise(ArgumentError, message: "invalid derivation path")
+      if nnum in @min_non_hardened_child_num..@max_non_hardened_child_num do
```

Comment:
Shouldn't this be @max_hardened_child_num? We want this block to cover all valid indexes.

Reply:
I don't think child num can be greater than 2^32 -1 when it's in a path string format

rust bitcoin https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/util/bip32.rs#L112

Reply:
Yes, sorry for the confusion.